### PR TITLE
UPSTREAM: <carry>: openshift: Remove dead code

### PR DESCRIPTION
**Pull request by ingvagabund**
We don't use it!!!

- `pkg/cloud/azure/actuators/cluster`
- No relevant code within `pkg/cloud/azure/actuators/machine`
- Gazelle generated files

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -39,6 +39,7 @@ required = [
     "sigs.k8s.io/controller-runtime/pkg/source",
     "sigs.k8s.io/testing_frameworks/integration", # for integration testing
     "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1",
+    "k8s.io/client-go/plugin/pkg/client/auth/gcp", # for cluster-api's cmd/manager
```
Comment (enxebre) [inline Gopkg.toml]: can you elaborate why we need this here?

**Comment by enxebre** [comment]
nice! it'd be nice if you could be more specific on the desc/commit about the bits that are being dropped

File: Gopkg.toml (hunk 1 above)
Comment (ingvagabund) [reply Gopkg.toml]: `# for cluster-api's cmd/manager`

**Comment by ingvagabund** [comment]
> nice! it'd be nice if you could be more specific on the desc/commit about the bits that are being dropped

Removing the code that we don't build/run/...

**Comment by enxebre** [comment]
```Removing the code that we don't build/run/...
```
Seems this is removing:
`pkg/cloud/azure/actuators/cluster`
`No relevant code within `pkg/cloud/azure/actuators/machine`
`Gazelle generated files`
Please prove me wrong. Can we update the desc with that?

**Pull request merged**